packages/web-api-scan-request-sender/src/sender/on-demand-scan-request-sender.ts

```diff
@@ -19,13 +19,20 @@ export class OnDemandScanRequestSender {
             onDemandPageScanRequests.map(async page => {
                 const resultDocs = await this.onDemandPageScanRunResultProvider.readScanRuns([page.id]);
                 const resultDoc = resultDocs.pop();
+                let isEnqueueSuccessful: boolean;
                 if (resultDoc !== undefined && resultDoc.run !== undefined && resultDoc.run.state === 'accepted') {
                     const message = this.createOnDemandScanRequestMessage(page);
-                    await this.queue.createMessage(this.storageConfig.scanQueue, message);
+                    isEnqueueSuccessful = await this.queue.createMessage(this.storageConfig.scanQueue, message);
+                }
+
+                if (isEnqueueSuccessful === true) {
                     await this.updateOnDemandPageResultDoc(resultDoc);
                 }
 
-                await this.pageScanRequestProvider.deleteRequests([page.id]);
+                if (isEnqueueSuccessful !== false) {
+                    // only when message queue failed we keep the request
+                    await this.pageScanRequestProvider.deleteRequests([page.id]);
```

Comment:
why not move this code to the above if block after the below line ?
                    await this.updateOnDemandPageResultDoc(resultDoc);

Reply:
Correct me if I’m wrong, if it has other states like queued we want to delete it as well. We have a test case for that.